main.go

```diff
@@ -1,392 +1,93 @@
-// Copyright (C) 2014 Jakob Borg and Contributors (see the CONTRIBUTORS file).
+// Copyright (C) 2014-2015 Jakob Borg and Contributors (see the CONTRIBUTORS file).
 
 package main
 
 import (
-	"bytes"
-	"encoding/binary"
+	"database/sql"
 	"flag"
-	"fmt"
-	"io"
 	"log"
 	"net"
 	"os"
-	"path/filepath"
-	"sync"
 	"time"
 
-	"github.com/golang/groupcache/lru"
-	"github.com/juju/ratelimit"
-	"github.com/syncthing/protocol"
-	"github.com/syncthing/syncthing/internal/discover"
-	"github.com/syndtr/goleveldb/leveldb"
-	"github.com/syndtr/goleveldb/leveldb/opt"
+	"github.com/thejerf/suture"
 )
 
-const cacheLimitSeconds = 3600
-
 var (
-	lock       sync.Mutex
-	queries    = 0
-	announces  = 0
-	answered   = 0
-	limited    = 0
-	unknowns   = 0
-	debug      = false
-	lruSize    = 1024
-	limitAvg   = 1
-	limitBurst = 10
-	limiter    *lru.Cache
+	lruSize     = 10240
+	limitAvg    = 5
+	limitBurst  = 20
+	globalStats stats
+	statsFile   string
+	backend     = "ql"
+	dsn         = "memory://discosrv"
```

Comment:
Ah, for the above, this would essentially only need to be `dns = getEnvDefault("DISCOSRV_DB_DSN", "memory://discosrv")` and we're good, I think.